src/eduid_common/session/session.py

```diff
@@ -441,7 +460,19 @@ def derive_key(app_key, session_key, usage, size):
     # the low number of rounds (3) is not important here - we use this to derive two keys
     # (different 'usage') from a single key which is comprised of a 256 bit app_key
     # (shared between instances), and a random session key of 128 bits.
-    return hashlib.pbkdf2_hmac('sha256', app_key, usage + session_key, 3, dklen = size)
+    if PY3:
```

Comment:
Handling of `str` between py3 and py3 is really annoying. I would do the following:

```python
if isinstance(usage, six.string_types):
  usage.encode('utf8')
```
which should be equivalent to:
```python
if type(usage) is not six.binary_type:
  usage.encode('utf8')
```

Reply:
On this subject, [`unicode_literals`][ul] is an interesting option to look into.

  [ul]: http://python-future.org/unicode_literals.html